src/RemoteRequest/TemporaryFileCachedRemoteGetRequest.php

```diff
@@ -0,0 +1,201 @@
+<?php
+
+namespace AmpProject\RemoteRequest;
+
+use AmpProject\Exception\FailedToGetCachedResponse;
+use AmpProject\RemoteGetRequest;
+use AmpProject\Response;
+use DateTimeImmutable;
+use Exception;
+
+/**
+ * Temporarily cache remote response.
+ *
+ * @package ampproject/amp-toolbox
+ */
+final class TemporaryFileCachedRemoteGetRequest implements RemoteGetRequest
+{
+
+    /**
+     * Prefix to use to identify cached file.
+     *
+     * @var string
+     */
+    const CACHED_FILE_PREFIX = 'amp-remote-request-';
+
+    /**
+     * Cache control header directive name.
+     *
+     * @var string
+     */
+    const CACHE_CONTROL = 'Cache-Control';
+
+    /**
+     * Default cache expiration time in seconds.
+     *
+     * The number represents one month time in seconds. It will be used by default for successful requests when
+     * the 'cache-control: max-age' was not provided.
+     *
+     * @var int
+     */
+    const EXPIRY_TIME = 2592000;
+
+    /**
+     * The decorated RemoteGetRequest instance.
+     *
+     * @var RemoteGetRequest
+     */
+    private $remoteGetRequest;
+
+    /**
+     * The directory abspath where the temporary file would be saved.
+     *
+     * @var string
+     */
+    private $directory;
+
+    /**
+     * The temporary file abspath.
+     *
+     * @var string
+     */
+    private $file;
```

Comment:
This is broken logic. The same instance of `TemporaryFileCachedRemoteGetRequest` can cache multiple different responses based on the URL that is passed into `get($url)`.

Therefore, this should not be a class property, but rather an argument that is forwarded by `get()`, as opposed to the `$directory` property which is constant for the entire lifetime of the object instance.

Otherwise, if any code would check the `$file` property without passing through the `get()` method, it would be in an inconsistent state.